Assets/_BForBoss/_Weapons/Scripts/WeaponBehaviour.cs

```diff
@@ -0,0 +1,117 @@
+using Sirenix.OdinInspector;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Perigon.Weapons
+{
+    public abstract class WeaponBehaviour : MonoBehaviour
+    {        
+        private const float FutherestDistanceToRayCast = 10000f;
+        private readonly Vector3 CenterOfCameraPosition = new Vector3(0.5f, 0.5f, 0);
```

Comment:
Should this be tied to `CrosshairBehaviour`? I think if we decide to implement any kind of crosshair snapping or something like that it will be pretty tightly coupled

Reply:
mmmm i would disagree with it being tied to `CrosshairBehaviour`, the position itself is related to the camera itself. Where as the crosshairBehaviour is more of the View of the Crosshair.

Let me know if that's what you're referring to

Reply:
Ah thats fair, that makes sense to me